src/lib/action.rs

```diff
@@ -115,6 +119,21 @@ pub struct Update<T: Clone> {
     label: String,
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```

Comment:
The `#[derive...` should be below the comment (I realise this is wrong in other parts of the code too!).